samples/bluetooth/bap_broadcast_source/src/main.c

```diff
@@ -33,6 +33,9 @@
 #include <zephyr/usb/usbd.h>
 
 BUILD_ASSERT(strlen(CONFIG_BROADCAST_CODE) <= BT_ISO_BROADCAST_CODE_SIZE, "Invalid broadcast code");
+BUILD_ASSERT(IN_RANGE(strlen(CONFIG_BROADCAST_NAME), BT_AUDIO_BROADCAST_NAME_LEN_MIN,
+		      BT_AUDIO_BROADCAST_NAME_LEN_MAX),
+	     "Invalid broadcast name");
```

Comment:
This is a good start, but it doesn't cover it completely. The broadcast name is maximum 32 characters, and this still allows 40 characters if using single-octet characters. 

I guess we need to define some sort of `utf8_char_cnt` or something in https://github.com/zephyrproject-rtos/zephyr/blob/main/lib/utils/utf8.c and use that to count the number of UTF-8 characters. 

That is outside the scope of this PR, so this is fine for now.